frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts

```diff
@@ -245,14 +245,19 @@ export const playerInspectorLogic = kea<playerInspectorLogicType>([
                     !!featureFlags[FEATURE_FLAGS.RECORDINGS_INSPECTOR_PERFORMANCE] &&
                     (tab === SessionRecordingPlayerTab.ALL || tab === SessionRecordingPlayerTab.PERFORMANCE)
                 ) {
-                    for (const event of performanceEvents || []) {
+                    const performanceEventsArr = performanceEvents || []
+                    for (let eventIndex = 0; eventIndex < performanceEventsArr.length; eventIndex++) {
+                        const event = performanceEventsArr[eventIndex]
                         const timestamp = dayjs(event.timestamp)
                         const responseStatus = event.response_status || 200
 
                         // NOTE: Navigtion events are missing the first contentful paint info so we find the relevant first contentful paint event and add it to the navigation event
                         if (event.entry_type === 'navigation' && !event.first_contentful_paint) {
-                            const firstContentfulPaint = (performanceEvents || []).find(
```

Comment:
Ah yeah didn't think about that. I wonder though, should we instead be linking based on the pageview_id property? I'm not sure how reliable it is to just take the first event after the navigation event...

Reply:
Oh good idea, I think that actually works really well. It also correctly accounts for navigations events that don't have corresponding paint events.

![Screenshot 2023-01-17 at 12 59 38 PM](https://user-images.githubusercontent.com/13460330/212976247-0b1b494a-352c-4a7d-84a6-f2cd56afedb3.png)